Q1:
a) 1534
b) 3131
c) 5313

Q2:
a)
void main() {
    Account a1 = new Account("James");
    Account a2 = new Account("William");
    Transaction t = new Transaction(a1, a2, 37.3);
    Bank b = new Bank();
    b.doTransfer(t);
}

b)
class Main {
    void main() {
        Addressee a = new Addressee("John", "13 Computing Drive");
        Parcel p = new Parcel(a);
        Courier c = new Courier(p);
        Depot d = new Depot();
        d.outForDelivery(c);
    }
}

c)
class Main {
    void main() {
        Library lib = new Library();
        Book bk = new Book("Book Title", "Sam");
        Borrower br = new Borrower("John", 1234);
        lib.loanBook(bk, br);
    }
}

Q3 (Assertions not tested)

Q4
a)
try {
    Main.loadFile();
} catch (IncorrectSizeException ex) {
    System.out.println("IncorrectSizeException");
} catch (IncorrectPermissionException ex) {
    System.out.println("IncorrectPermissionException");
} catch (FileMetadataException ex) {
    System.out.println("FileMetadataException");
} catch (FileNotFoundException ex) {
    System.out.println("FileNotFoundException");
} catch (FileIOException ex) {
    System.out.println("FileIOException");
}

b)
try {
    Main.accessDatabase();
} catch (NoPermissionException ex) {
    System.out.println("NoPermissionException");
} catch (WriteException ex) {
    System.out.println("WriteException");
} catch (ElementNotFoundException ex) {
    System.out.println("ElementNotFoundException");
} catch (ReadException ex) {
    System.out.println("ReadException");
} catch (TableException ex) {
    System.out.println("TableException");
} catch (DatabaseException ex) {
    System.out.println("DatabaseException");
}